src/navsat_transform.cpp

```diff
@@ -310,6 +320,14 @@ namespace RobotLocalization
 
       utmWorldTransInverse_ = utmWorldTransform_.inverse();
 
+      // publish the origin of the world frame in NavSatFix format
+      sensor_msgs::NavSatFix nsf;
+      NavsatConversions::UTMtoLL(utmWorldTransInverse_.getOrigin().getY(), 
```

Comment:
You can't return references that were instantiated inside of a function because it is created in memory that is in a temporary (on the stack) which is de-allocated when the function ends, and it causes a segfault if you try to access that memory location. (This is how you can hijack services on servers)

You can however return global references, parameter references, or a copy of the value in the register/memory. 

Most languages that don't explicitly state that the parameter is a reference is actually a reference behind the scenes.

In languages where you're "returning" a value, if it is a variable that was instantiated inside the function then you will be returning an entire copy of the data structure which is super slow (some compilers can optimize against this).